test/Monad.hs

```diff
@@ -0,0 +1,31 @@
+{-# LANGUAGE RankNTypes #-}
+module Monad
+( test
+) where
+
+import Control.Monad ((>=>), ap)
+import Gen
+import Test.Tasty
+import Test.Tasty.Hedgehog
+
+test
+  :: (Monad m, Arg a, Arg b, Eq (f a), Eq (f b), Eq (f c), Show a, Show b, Show c, Show (f a), Show (f b), Show (f c), Vary a, Vary b, Functor s)
+  => (forall a . Show a => Gen a -> Gen (m a))
+  -> Gen a
+  -> Gen b
+  -> Gen c
+  -> Gen (s ())
+  -> (forall a . s (m a) -> PureC (f a))
```

Comment:
Passing in a generator for an input functor `s` here allows us to test initial state for carriers like `StateC` and `FreshC`. This accommodates arbitrary functors, but pairs are the only ones we use right now.

Taking a generator for the unit state allows us to avoid having to pass in a constructor function building input state from arbitrary parameters; this is the same trick as is used by `Effect` to pass the initial state separately from the effect.